src/common/main/kotlin/tech/thatgravyboat/skyblockapi/api/remote/api/RepoEnchantmentApi.kt

```diff
@@ -0,0 +1,38 @@
+package tech.thatgravyboat.skyblockapi.api.remote.api
+
+import net.minecraft.core.component.DataComponents
+import net.minecraft.world.item.ItemStack
+import net.minecraft.world.item.Items
+import net.minecraft.world.item.component.ItemLore
+import tech.thatgravyboat.repolib.api.RepoAPI
+import tech.thatgravyboat.skyblockapi.utils.builders.ItemBuilder
+import tech.thatgravyboat.skyblockapi.utils.extentions.*
+import tech.thatgravyboat.skyblockapi.utils.text.Text
+import tech.thatgravyboat.skyblockapi.utils.text.Text.asComponent
+
+object RepoEnchantmentApi {
+    private val cache: MutableMap<String, ItemStack?> = mutableMapOf()
+
+    fun getEnchantmentAsItemOrNull(id: String, level: Int) = cache.getOrPut(id) {
```

Comment:
This is cached with the id but uses the level in the itemstack. This should also allow level to be nullable to grab the highest level of the enchantment